lib/jekyll/filters/url_filters.rb

```diff
@@ -10,6 +10,12 @@ module URLFilters
       # Returns the absolute URL as a String.
       def absolute_url(input)
         return if input.nil?
+        if Addressable::URI.parse(input).absolute?
+          Jekyll.logger.warn(
+            "absolute_url:",
+              %("#{input}" appears to already be absolute)
+          )
```

Comment:
From a usability standpoint, the first thing I think about is: how is the user going to have any idea *where* in their site to look? If you don't think it should be automatically dealt with, then we should `raise` or something so we can get some context.